src/js/components/meta/FormiflyForm.tsx

```diff
@@ -1,17 +1,19 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import {ThemeProvider} from 'styled-components';
 import {FormiflyProvider, useFormiflyContext} from './FormiflyContext';
+import {SubmitFunction, SubmitValidationErrorFunction, ValueOfValidator} from '../../types';
+import BaseValidator from '../../classes/BaseValidator';
 
-const Form = (props) => {
-    const {handleSubmit} = useFormiflyContext();
+const Form = <T extends BaseValidator<any>>(props: FormProps<T>) => {
+    const {handleSubmit} = useFormiflyContext<T>();
     return <form onSubmit={e =>
-        handleSubmit(props.onSubmit, props.onSubmitValidationError, e)} className={'formifly-form ' + (props.className ?? '')}>
+        handleSubmit(props.onSubmit, props.onSubmitValidationError, e)}
+                 className={'formifly-form ' + (props.className ?? '')}>
         {props.children}
     </form>;
 };
 
-const FormiflyForm = (props) => {
+const FormiflyForm = <T extends BaseValidator<any>>(props: FormiflyFormProps<T>) => {
```

Comment:
why does the form type extend BaseValidator?

Reply:
It doesn't. The generic type T does and FormiflyForm depends on that, but the form itself doesn't extend the Validator. This is basically 'calculating' the type for Values and UnpackedErrors depending on the shape you pass to Formifly. I think this is easier explained in a call